Update updating OpenTelemetry Collector dependencies doc (#2049)
grafana/alloy: docs/developer/updating-otel/README.md

You can refer to [PR grafana/agent#5290](https://github.com/grafana/agent/pull/5290)
for an example on how to update Alloy.

### Notify community component contributors

You can find their GitHub handles at the top of the community component files (the components that have "Community" set to true).
Once the PR is created, you can ping them on it and message them on Slack. Each contributor should create a PR to your branch with the updated component.

> They should be notified a few days before the release so that they have enough time to do the update.

Reviewer: ```suggestion
Notify the contributors a few days before the release so they have enough time to do the update.
```

> Notify the contributors a few days before the release so they have enough time to do the update.

Reviewer 2: Most of the time when you read these docs you will be doing the release, so I'm not sure if it's the right place for a reminder like this.

Author: this is the "updating-otel" doc. The update should be done prior to the release and should be started at least one week before cutting the RC

> It should not be a blocker for the release. If they can't do the update, you can fix the code to resolve the breaking changes or disable the component if that's too complicated.

Reviewer: ```suggestion
It should not be a blocker for the release. If the contributors can't do the update, you can fix the code to resolve the breaking changes or disable the component if that's too complicated.
```
